### ISSUE TITLE
BelongsTo seems not update canonicalRecord when it is set to null

### ISSUE THREAD

**Issue by SelokYeung** [emberjs/data#3640]
I was trying to rollback a relationship, and I found that canonicalRecord didn't update after save if it's value is set to null.  I don't know whether it is issue, but it is weird for canonicalRecord still keep the invalid record after save.

[data/packages/ember-data/lib/system/relationships/state/belongs-to.js](https://github.com/emberjs/data/blob/v1.13.7/packages/ember-data/lib/system/relationships/state/belongs-to.js#L34)

I guess it should be modify as follow
```javascript
BelongsToRelationship.prototype.setCanonicalRecord = function (newRecord) {
	if (newRecord) {
		this.addCanonicalRecord(newRecord);
	} else {
		this.removeCanonicalRecord(this.canonicalState);
		if (this.inverseRecord) {
			this.removeCanonicalRecord(this.inverseRecord);
		}
	}
	this.setHasData(true);
};
```

**Comment by gilpinheiro** [comment]
I'm seeing the same thing, this has bitten me. ember-data: 1.13.4

**Comment by SelokYeung** [comment]
I do it in dirty ways, I hooked didUpdate event on every model. 
So everytime those model saved. It will check for all belongsto relationship. if it is empty then cleanup canonical record.

```javascript
onDataChanged: function () {
	var self = this;

	this.eachRelationship(function (name, relationship) {
		if (relationship.kind === 'belongsTo') {
			if (Em.isEmpty(self.get(name))) {
				var internalRelation = self.get('_internalModel')._relationships.get(name);
				internalRelation.removeCanonicalRecord(internalRelation.canonicalState);
			}
		}
	});
}.on('didUpdate'),
```

Seems work fine in ember-data: 1.13 and 2.0

**Comment by bmac** [comment]
Resolved via 8400b9ddbce36f7d93c5b519eda974e46f612fd8
